lib/compress.js

```diff
@@ -261,6 +262,24 @@ merge(Compressor.prototype, {
                         d.fixed = false;
                     }
                 }
+                if (node instanceof AST_Defun) {
+                    var d = node.name.definition();
+                    if (!toplevel && d.global || is_safe(d)) {
+                        d.fixed = false;
+                    } else {
+                        // Make a shallow copy of definition body but as
+                        // expression instead - ensure to clone whatever
+                        // child nodes you extract from this
```

Comment:
Thanks for the advice 👍 

Done in https://github.com/mishoo/UglifyJS2/pull/1560/commits/0d99a451c77c4786af06c1507648d13776e68121